server/datastore/mysql/policies.go

```diff
@@ -283,63 +286,151 @@ func (ds *Datastore) RecordPolicyQueryExecutions(ctx context.Context, host *flee
 	return nil
 }
 
-func (ds *Datastore) ListGlobalPolicies(ctx context.Context) ([]*fleet.Policy, error) {
-	return listPoliciesDB(ctx, ds.reader(ctx), nil, nil)
+func (ds *Datastore) ListGlobalPolicies(ctx context.Context, opts fleet.ListOptions) ([]*fleet.Policy, error) {
+	return listPoliciesDB(ctx, ds.reader(ctx), nil, nil, opts)
 }
 
 // returns the list of policies associated with the provided teamID, or the
 // global policies if teamID is nil. The pass/fail host counts are the totals
 // regardless of hosts' team if countsForTeamID is nil, or the totals just for
 // hosts that belong to the provided countsForTeamID if it is not nil.
-func listPoliciesDB(ctx context.Context, q sqlx.QueryerContext, teamID, countsForTeamID *uint) ([]*fleet.Policy, error) {
+func listPoliciesDB(ctx context.Context, q sqlx.QueryerContext, teamID, countsForTeamID *uint, opts fleet.ListOptions) ([]*fleet.Policy, error) {
 	var args []interface{}
 
+	var initialQuery string
+
+	// Sorting by failing host counts requires an expensive join with the
+	// policy membership table and may result in long response times
+	if opts.OrderKey == "failing_host_count" {
+		if countsForTeamID != nil {
+			initialQuery = `
+				SELECT p.id
+				FROM policies p
+				LEFT JOIN (
+					SELECT pm.policy_id,
+						COUNT(*) AS failing_host_count
+					FROM policy_membership pm
+					INNER JOIN hosts h ON pm.host_id = h.id AND pm.passes = false AND h.team_id = ?
```

Comment:
Maybe better to `INNER JOIN policies` instead of `hosts`? (`policies` has `team_id` too).